task1: RP intermediary use case (#118)
webuild-consortium/wp4-trust-group: task1-use-cases/subtask1-2-trust-registry/relying-party-intermediary-use-case.md

# Use Case UC-RPI-01: Relying Party Intermediary

## Scope

An **Intermediary** is a Relying Party that requests attribute presentation from a Wallet Unit **on behalf of** another registered Relying Party (the **intermediated RP**). Typical examples: identity/interop gateways, sector platforms, or SaaS verifiers that serve multiple service providers.

This use case covers registration, intermediated remote presentation (OpenID4VP), and wallet-side trust evaluation. Protocol details and JSON examples: [RPI OpenID4VP technical report](../../task2-trust-framework/rp-intermediary-openid4vp-technical-report.md).

## Actors

| Actor | Role |
| ----- | ---- |
| **Intermediary** | Registers as RP with `isIntermediary`; holds **WRPAC**; sends presentation requests |

> | **Intermediated RP** | Registered at a Registrar; may hold **WRPRC** per intended use; instructs intermediary what to request |

Reviewer: ```suggestion
Intermediated WRP | A Wallet Relying Party registered at a Registrar whose intended uses may be registered by itself and/or by one or more intermediaries on its behalf. For each intended use a WRPRC may be issued at the requester's request; it states what may be requested for that intended use and links to the intermediary where one registered it.
```

Author: WRPRC is now mandatory, according to ARF 3.0 and CIR (EU) 2026/1730, below further details in support of this update:

RPRC_19 requires a Relying Party Instance to put a single WRPRC in every presentation request, by value, for the current Service and intended use (proximity and remote).

RPRC_17 requires the Wallet to verify that WRPRC. If it is missing, malformed, inauthentic, or expired, the Wallet warns the User at approval. Whether the User may still approve is a Wallet Provider risk-policy decision, not an automatic refusal.

Wallet verification of WRPRCs is also deferred: ARF v3.0.0 notes that RPRC_17 applies only 24 months after the regulation amending CIR 2024/2982 enters into force.

There is no user-opt-in Registrar lookup instead of a missing WRPRC (RPRC_16 / RPRC_18 are empty).

for further details, please see PR https://github.com/webuild-consortium/wp4-trust-group/pull/126

## Scenarios

| ID | Scenario | Intermediated? |
| -- | -------- | -------------- |
| **S1** | Interop gateway requests PID for a cinema booking service | Yes |
| **S2** | Intermediary requests attributes for its **own** service | No (direct RP) |

### S1 — Intermediated presentation (primary)

**Example:** *VerifyNow* (intermediary) requests age verification on behalf of *Example Cinema Ltd.* (intermediated RP).

**Preconditions**

- Intermediary registered with `isIntermediary: true` and valid **WRPAC**.
- Intermediated RP registered; intended use and optional **WRPRC** issued; `usesIntermediary` points to the intermediary (RPI_04).
- Intermediated RP provided the intermediary with: trade name, identifier, Registrar URL, intended-use id, attribute list, optional WRPRC reference (RPI_05).

**Main flow**

```mermaid
sequenceDiagram
    participant IRP as Intermediated RP
    participant INT as Intermediary
    participant W as Wallet Unit
    participant U as User

    IRP->>INT: Request attributes (RPI_05 metadata)
    INT->>W: OpenID4VP Request Object<br/>(INT WRPAC + IRP verifier_info)
    W->>W: Detect intermediation (RPI_07)<br/>Validate INT WRPAC + IRP WRPRC/registry

>     W->>U: Show INT and IRP names; request consent

Reviewer 2: fix chart 
```suggestion
    W->>U: Show Intermediated RP names + ask user for consent
```

    U->>W: Approve or deny
    W->>INT: Authorization Response (response_uri)

>     INT->>IRP: Forward attributes (RPI_08/09); delete after use (RPI_10)

Reviewer 2: fix chart (splitting the row in 2 ) 
```suggestion
    INT->>IRP: Forward attributes (RPI_08/09) 
    INT->>INT: delete attributes after use (RPI_10)
```

```

1. Intermediated RP instructs the intermediary (RPI_05).
2. Intermediary builds and signs the OpenID4VP Request Object with its **WRPAC**; includes intermediated RP `registrar_dataset` and optional `registration_cert` in `verifier_info` (RPI_06, RPRC_19a).

> 3. Wallet detects intermediation when WRPAC `organizationIdentifier` ≠ `registrar_dataset.identifier` (RPI_07).

Reviewer 2: rephrase 
```suggestion
3. Wallet detects an intermediated presentation when WRPAC `organizationIdentifier` ≠ `registrar_dataset.identifier` (RPI_07).
```

4. Wallet validates intermediary WRPAC (Access CA TL) and intermediated RP WRPRC/registry entitlements ([UC-TE-04](wallet-unit-evaluates-relying-party.md)).
5. Wallet displays **both** trade names; User approves (RPA_07).
6. If User opted to verify registration, Wallet checks intermediary–RP relationship via WRPRC `intermediary` field (RPRC_04) or Registrar API (RPI_07a).
7. Wallet sends response to intermediary `response_uri`; intermediary forwards to intermediated RP.

**Postconditions (success)**

> - User saw both intermediary and intermediated RP before consent.

Reviewer 2: in my opinion this is incorrect. 
The user does not see both intermediary and intermediated RP information before giving consent. 
See note in 6.6.5 of ARF 
`Note: The Wallet Unit does not display the trade names of the intermediary and its Service to the User, because the intermediary operates on behalf of the intermediated Relying Party and is not a party in the use case taking place between the User and the Relying Party. The User would probably not expect or recognise the name of the intermediary or its Service, and may be confused if the Wallet Unit would show these names. However, the intermediary name and service are logged by the Wallet Unit, see [Section 6.6.3.13](https://eudi.dev/3.0.0/main/06-trust-model/#66313-wallet-unit-enables-the-user-to-report-suspicious-requests-by-a-relying-party-and-to-request-a-relying-party-to-erase-personal-data), so the fact that an intermediary was used in the transaction is visible to the Use. ` 

Proposing the following suggestion

```suggestion
- The wallet does not display the information of the intermediary and its Service to the User before asking for  consent. Only the information of the intermediated RP are shown. The intermediary name and service are logged by the Wallet Unit. 
```

Reviewer: I agree

Author: yeah, according to ARF 3.0

- The Wallet *SHALL NOT* display the trade names of the *intermediary* and the *intermediary Service* when asking for User approval (*RPA_07* / *RPA_06* note b).
- The Wallet displays the *intermediated Relying Party* (and its Service) from the WRPRC in the request.
- Intermediation is detected when the WRPAC subject is the intermediary and the WRPRC identifies a different entity that uses that intermediary (*RPRC_17a*).

- Attributes disclosed only match intermediated RP registered entitlements (RPRC_21).
- Intermediary deletes received credentials immediately after forwarding (RPI_10).

### S2 — Intermediary as direct Relying Party

An entity registered as intermediary may also act **in its own capacity** (RPI_01 note c): WRPAC and `verifier_info` refer to the **same** party. Flow matches [UC-TE-04](wallet-unit-evaluates-relying-party.md) without dual-identity handling.

## Onboarding (summary)

> Detailed steps: [Relying Party Onboarding](../subtask1-1-onboarding/relying_party_onboarding.md).

Reviewer 2: Suggesting the following update 

```suggestion
Both the Intermediary and the Intermediated RP are considered RP. The onboarding process shall follow the detailed steps in: [Relying Party Onboarding](../subtask1-1-onboarding/relying_party_onboarding.md).
```

Need more time to review Relying Party Onboarding to see if any update is needed there

> 

Reviewer 2: ```suggestion
The following table highlight the on-boarding step differences between Intermediary and Intermediated RP
```

| Step | Intermediary | Intermediated RP |
| ---- | ------------ | ---------------- |
| Register | As RP with `isIntermediary: true` (RPI_01, Reg_26) | At Registrar in establishment MS (RPI_03) |
| Relationship | — | Provide evidence of intermediary use; Registrar sets `usesIntermediary` (RPI_04) |
| Certificates | **WRPAC** (mandatory); own WRPRC only for direct use (S2) | **WRPRC** per intended use; includes `intermediary` association (RPRC_04) |
| Registry API | May register intended uses on behalf of WRP ([TS5 notes](../../task5-participants-certificates-policies/ts5-registry-api-and-data-formats.md)) | — |

**Note:** Intermediated RPs do **not** need a WRPAC (ARF §6.6.5).

## Wallet evaluation checklist

| Check | Source | Applies to |
| ----- | ------ | ---------- |

> | WRPAC chain + revocation | Access CA TL | Intermediary |

Reviewer 2: I propose to add the following checks: 

- Intermediation Detection
- Request Object Signature 

```suggestion
| WRPAC chain + revocation | Access CA TL | Intermediary |
| Intermediation Detection | WRPAC from intermediary differs from verifier_info | Both|
| Request Object Signature | public key contained in x5c | Intermediary |
| WRPAC chain + revocation | Access CA TL | Intermediary |
```

| WRPRC signature + entitlements | WRPRC Provider TL / registry | Intermediated RP |
| Requested attributes ⊆ registered | RPRC_21 | Intermediated RP |

> | Dual identity displayed | RPI_07 | Both |

Reviewer 2: ```suggestion
| Only Intermediated RP displayed | RPI_07 | Intermediated RP|
```

> | Relationship registered | RPI_07a, RPRC_04 | Intermediary ↔ IRP |

Reviewer 2: Starting from ARF v2.9.0 and following version **RPI_07** has been removed. 

In the ARF 2.8.0 RPI_07 cited : 
`In case a Wallet Unit receives a presentation request from an intermediary on behalf of an intermediated Relying Party, and if the User indicated that they want to verify the information registered about this Relying Party (according to RPRC_16), the Wallet Unit SHALL verify that that the contractual relationship between the Relying Party and the intermediary is indeed registered by the responsible Registrar according to RPI_04, see also RPRC_04. If this verification fails, the Wallet Unit SHALL notify the User when asking for User consent. Note: The Wallet Unit can either do this by inspecting the registration certificate (if available) or by querying the Registrar.`
 
From ARF v2.9.0 section [6.6.5 PID or attestation presentation to an intermediary](https://eudi.dev/2.9.0/main/06-trust-model/#665-pid-or-attestation-presentation-to-an-intermediary)
the following is stated 
`1. If the Wallet Unit sees in the registration certificate that the Relying Party uses the services of an intermediary, it verifies that the identifier and Service identifier of the intermediary listed in the usesIntermediary field of the registration certificate are identical to the identifier and Service identifier listed in the access certificate` 

I suggest to modify the source column as follows. 

```suggestion
| Relationship registered | 6.6.5 ARF | Intermediary ↔ Intermediated RP |
```

please note that RPI_07a is flagged as empty in arf 3.0.0

Reviewer: +1

| EDP (if present) | EDP_02/03 | Evaluate against **intermediated** RP id/root, not intermediary WRPAC |

## Success criteria

- Intermediary authenticated via valid WRPAC.
- Intermediated RP identity and entitlements verified (WRPRC or registry).

> - User informed of both parties before consent.

Reviewer 2: ```suggestion
- User informed only of Intermediated RP informations before consent.
```

- Intermediated transaction does not expose intermediary WRPRC in `verifier_info`.

## ARF requirements (key)

| ID | Summary |
| -- | ------- |
| RPI_01 | Intermediary registers as RP; obtains WRPAC |
| RPI_03 | Intermediary registers each intermediated RP |
| RPI_04 | Registrar verifies and records intermediary relationship |
| RPI_05 | Intermediated RP supplies request metadata to intermediary |
| RPI_06 | Intermediary sends request with own WRPAC + IRP data/WRPRC |

> | RPI_07 | Wallet detects and displays both identities |

Reviewer 2: ```suggestion
| RPI_07 | Wallet detects and displays only intermediated RP identity |
```

> | RPI_07a | Wallet verifies registered relationship when User requests |

Reviewer 2: I think that this check is still valid but starting from ARF 2.9.0 RPI_07a is missing but the check is included in section [6.6.5 PID or attestation presentation to an intermediary](https://eudi.dev/2.9.0/main/06-trust-model/#665-pid-or-attestation-presentation-to-an-intermediary)

I suggest to refer to Topic 52 where RPI_07a was included. 

```suggestion
| Topic 52  | Wallet verifies registered relationship when User requests |
```

| RPI_08–10 | Forwarding, verification, immediate deletion by intermediary |
| RPRC_04 | Intermediated WRPRC contains intermediary association |

> | RPRC_19a | Registration data in presentation request extension |

Reviewer 2: RPRC_19a has been removed starting from ARF 2.9.0

I suggest to delete it. 

```suggestion
```
